src/options.cpp

```diff
@@ -364,6 +394,12 @@ void cOpt::setNext()
 
         sSet = vItems[iNext];
 
+    } else if (sType == "string_input") {
+        int iMenuTextLength = sMenuText.length();
+        sSet = string_input_popup("", (iMaxLength > 80) ? 80 : ((iMaxLength < iMenuTextLength) ? iMenuTextLength : iMaxLength+1),
```

Comment:
Right, an empty string is a valid value for that option, the user may enter it on purpose. In that case, it's probably not so easy to fix. I know other code relies on this cancel==empty string behavior.

The idiomatic way would be to return a `std::pair<std::string, bool>`, similar to what `std::map::insert` returns (the boo indicates whether the user has canceled the input).